api/api_test.go

```diff
@@ -1119,6 +1119,65 @@ func TestAPI_GenerateEnvHTTPS(t *testing.T) {
 	require.Equal(t, expected, c.GenerateEnv())
 }
 
+// TestAPI_PrefixPath() validates that Config.Address is split into
+// Config.Address and Config.PathPrefix as expected.  If we want to add end to
+// end testing in the future this will require configuring and running an
+// API gateway / reverse proxy (e.g. nginx)
+func TestAPI_PrefixPath(t *testing.T) {
+	t.Parallel()
+
+	// reverse proxy PathPrefix should be separated out
```

Comment:
I'm curious if this might be simpler to express with a table of inputs and expected outputs? This is basically what this is already doing except for the examples without a prefix are separated below. If we defined a `testCase` type that looked like:
```
type testCase struct {
    name   string
    addr   string
    expect string
}
```

You could define all your cases like:
```
cases := []testCase{
    {
        name: "with http and prefix",
        addr: ""http://reverse.proxy.com/consul/path/prefix",
        expectAddr: "reverse.proxy.com",
        expectPrefix: "/consul/path/prefix",
    },
    ...
}
```
and then loop over cases like:
```
for _, tc := range cases {
    t.Run(t, func(t *testing.T) {
        c := &Config{Address: tc.addr}
        client, err := NewClient(c)
        require.NoError(t, err)
        require.Equal(t, tc.expectAddr, client.config.Address) # using the require package here which is used elsewhere
        require.Equal(t, tc.expectPrefix, client.config.PathPrefix)
    })
}
```

Something like that? We follow a pattern like this in a lot of our tests (example [here](https://github.com/hashicorp/consul/blob/main/api/agent_test.go#L1794-L1828) and [here](https://github.com/hashicorp/consul/blob/main/api/health_test.go#L60-L191)) when it's easy to express input/output cases. At the very least, it makes the test code little easier to read all at once, and a little more concise, but not necessarily "better" than what you have already. 

If that makes sense to you cool, if not, it's just as suggestion. I just spat that out and didn't try to run it or anything but should illustrate what I'm talking about. Take from it what you will.

Reply:
sure, I'll do that and push an update.

Reply:
done